### ISSUE TITLE
Bad aggregation name complains about "unknown BaseAggregationBuilder" instead of "unknown Aggregation"

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#28950]
If you write an aggregation as

```
           POST /vehicles/_search
           {
	              "aggs": {
 	                       "myname": {
        		                 "missing": {
        		                 "field": "make.keyword"
        		                 }
    	                        }
	               }
             }
```

Then we say `Unknown BaseAggregationBuilder [missing]`. It'd be *much* more user friendly to say "Unknown Aggregation [missing]."

**Comment by colings86** [comment]
I agree, we should fix this.

**Comment by nik9000** [comment]
Thanks for fixing my missing ping @hub-cap !

**Comment by nikhilbarar** [comment]
Is this issue fixed?

**Comment by talevy** [comment]
@nikhilbarar not yet. It will most likely be closed and associated to the Pull Request that implements the fix itself.

**Comment by liketic** [comment]
Hi @nik9000 When I trying to reproduce this on master, I got:
```
{
  "took": 2,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 0,
    "max_score": null,
    "hits": [
      
    ],
    
  },
  "aggregations": {
    "myname": {
      "doc_count": 0
    }
  }
}
``` .  How can I reproduce this bug? Thanks in advance.

**Comment by nik9000** [comment]
Oops! I meant to use `missing` as an example of an aggregation that doesn't exist. But it does exist! Here is a better example:
```
curl -HContent-Type:application/json -XPOST localhost:9200/vehicles/test?pretty -d'{"test":"test"}'
curl -HContent-Type:application/json -XPOST localhost:9200/vehicles/_search?pretty -d'{
    "aggs": {
        "myname": {
            "agg_that_does_not_exist": {
                "field": "make.keyword"
            }
        }
    }
}'
```

**Comment by nik9000** [comment]
Fixed by #58255. #29201 has another fix that I think is actually better. Let's see if we can get that one in too!
